ros2debug/ros2debug/api/__init__.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import platform
+
+
+# Installation system requirements of ROS 2
+ROS2_REQS = {'crystal': {'linux': {'version': 'Ubuntu',
+                                   'release': ['16.04', '18.04'], 
+                                   'processor': 'x86_64'},
+                        'darwin': {'version': '',
+                                   'release': ['15.0.0', '15.6.0',
+                                               '16.0.0', '16.5.0',
+                                               '16.6.0'],
+                                   'processor': ''},
+                        'windows': {'version': '10',
+                                    'release': [],
+                                    'processor': 'AMD64'}
+                        },
+             'dashing': {'linux': {'version': 'Ubuntu',
+                                  'release': '18.04',
+                                  'processor': 'x86_64'},
+                        'darwin': {'version': '',
+                                   'release': ['14.0.0', '14.5.0', '15.0.0',
+                                               '15.6.0', '16.0.0', '16.5.0',
+                                               '16.6.0', '17.0.0', '17.5.0',
+                                               '17.6.0', '17.7.0'],
+                                   'processor': ''},
+                        'windows': {'version': '10',
+                                    'release': [],
+                                    'processor': 'AMD64'}
+                        }
+            }
```

Comment:
Embedding this kind of information in the code requires it to be constantly updated. I am not sure that is somthing we want to do. Some information (like the status of a ROS distribution) can be queries through `rosdistro` - e.g. https://github.com/ros/rosdistro/blob/87bd1a05b0294b55554d9e339e178e166227c7ee/index-v4.yaml#L29. Atm we only store the Debian platforms (https://github.com/ros/rosdistro/blob/87bd1a05b0294b55554d9e339e178e166227c7ee/dashing/distribution.yaml#L5) but maybe it would be a good idea to enumerate the other platforms too to make them machine readable.

Reply:
I will take a look and change that part.